src/app/modules/submission/encrypt-submission/encrypt-submission.controller.ts

```diff
@@ -57,11 +58,28 @@ const EncryptSubmission = getEncryptSubmissionModel(mongoose)
 // NOTE: Refer to this for documentation: https://github.com/sideway/joi-date/blob/master/API.md
 const Joi = BaseJoi.extend(JoiDate)
 
-const submitEncryptModeForm: RequestHandler = async (req, res) => {
+const submitEncryptModeForm: RequestHandler<
+  { formId: string },
+  {
+    message: string
+    submissionId?: string
+    spcpSubmissionFailure?: boolean
+    isPageFound?: boolean
+    formTitle?: string
+  },
+  {
+    responses: FieldResponse[]
+    encryptedContent: string
+    isPreview: boolean
+    attachments?: Record<string, unknown>
+    version: number
+  },
+  { captchaResponse?: unknown }
```

Comment:
I reckon it would help future developers if we gave meaningful names to these types, instead of anonymously inlining them here

Reply:
Any examples? I've changed part of this to use `EncryptSubmissionDto` instead. Should I do similar things for the rest of them?

Reply:
yes that would help!